Makefile

```diff
@@ -86,15 +86,10 @@ PINNED_DEPENDENCIES := \
 	github.com/go-sql-driver/mysql@v1.5.0 \
 	github.com/urfave/cli/v2@v2.4.0
 
-# Code coverage output files.
+# Code coverage & test report output files.
 COVER_ROOT                      := ./.coverage
-UNIT_COVER_PROFILE              := $(COVER_ROOT)/unit_coverprofile.out
-INTEGRATION_COVER_PROFILE       := $(COVER_ROOT)/integration_coverprofile.out
-DB_TOOL_COVER_PROFILE           := $(COVER_ROOT)/db_tool_coverprofile.out
-FUNCTIONAL_COVER_PROFILE        := $(COVER_ROOT)/functional_$(PERSISTENCE_DRIVER)_coverprofile.out
-FUNCTIONAL_XDC_COVER_PROFILE    := $(COVER_ROOT)/functional_xdc_$(PERSISTENCE_DRIVER)_coverprofile.out
-FUNCTIONAL_NDC_COVER_PROFILE    := $(COVER_ROOT)/functional_ndc_$(PERSISTENCE_DRIVER)_coverprofile.out
```

Comment:
I've inlined these inline the respective task instead.

Reply:
Consider using this trick? https://github.com/temporalio/temporal/pull/4589/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R96

Reply:
I like that! The only reason I'm hesitating is that the way it's currently set up - by using the script `develop/buildkite/scripts/coverage-report.sh` which downloads the coverage data from each step explicitly - is actually useful right now to prevent uploading partial coverage data. That's because it'll fail if _any_ of the test steps failed. That's now useful because I had to add `continue_on_failure: true` to the `wait` step at the end due to the fact that we are now actually interested in reporting on build failures for the analytics. After reading the Buildkite docs, so far I couldn't figure out how to replicate that behaviour.

Reply:
Hm, actually I could change the script to use a pattern, like `buildkite-agent artifact download .coverage/*_coverprofile.out` instead. But then I'd still need that script and I'm not sure I've gained _a lot_.

Reply:
I deleted the script in my branch and just download with a wildcard. Is uploading partial coverage data really a big deal?

Reply:
Actually, I think this might work with an additional `wait`; let me try that.